content/en/blog/2022-08-05-effective-programmers-read-code.md

```diff
@@ -0,0 +1,59 @@
+
+The more code we read the quicker and more effective we are at building the glass castle, the better code we will write. To become effective authors of good code, we should be first avid readers of good code.
+
+#### P.S. Tips for code readers
+
+Reading code must also be a practice that we focus on and we should prepare ourselves for success.
+
+1. _Ensure you have long periods of uninterrupted development time._ Give yourself as much time with the glass castle as possible. At Rotational we try to schedule all of our meetings for Tuesdays and Thursdays to leave Monday and Wednesday for deep work.
+2. _Create shortcuts to build the glass castle faster._ If you can create mental checkpoints, notes to yourself, or other mnemonics you will avoid treading the same path over and over again.
+3. _Diversify the code you are reading._ Don't just read the code that you or your colleagues wrote, or even stick to code in the language you're working in. There are a huge number of excellent open source projects that you can read; make this practice part of your continuous learning strategy.
+4. _Glass castle before code review._ The glass castle strategy isn't just about implementing code, it's also about reviewing code.
+5. _Read code to write documentation._ It is difficult to make documentation part of the definition of done of a task, instead create tasks for documentation that allow you to read code from a user or test developers perspective.
+
+#### P.S. Writing for those who read code
+
+We started this essay by discussing the two audiences for code: the runtime and the reader. The runtime is the execution environment of the code including the compiler or the interpreter, other code that imports or loads your code, and the operating system that runs your code in a process. The reader is other programmers: users of your code as a module, developers who are writing tests for your code, maintainers who deploy code, fix bugs and improve performance, and future you who has the expertise to understand what, how, and why of the source code.
+
+The first and most important notion is that these audiences are themselves dynamic. The more time that passes from the original implementation of the source code, the more different both the runtime and the reader are. We should write code to ensure longevity of executability and understandability.
+
+When it comes to writing code, we certainly should not only focus on or prioritize execution performance if it sacrifices readability. Self documenting code is absolutely something to strive for. On the other hand, clean coders eschew comments and have probably have taken this to the extreme. Obtuse code serves no one, not future readers and not compilers. Verbose comments that repeat the code make it harder to build the glass castle. Instead, use comments to create markers or way points that help guide the reader to other pieces of important or related code, or that serve to build the glass castle more efficiently.
```

Comment:
Suggestion: "On the other hand, some clean coders who comment in excess have probably have taken this to an extreme."

"code, or places that serve to build the glass castle more efficiently."